File: Server/src/main/java/bgu/spl171/net/impl/TFTP/CommandBuilders/LoginRequestCommandBuilder.java
package bgu.spl171.net.impl.TFTP.CommandBuilders;

import bgu.spl171.net.impl.TFTP.Commands.Command;
import bgu.spl171.net.impl.TFTP.Commands.LoginRequest;
import bgu.spl171.net.impl.TFTP.Commands.ReadRequest;
import bgu.spl171.net.impl.TFTP.EncoderDecoders.StringEncoderDecoder;

public class LoginRequestCommandBuilder implements CommandBuilder {
    StringEncoderDecoder stringEncoderDecoder = new StringEncoderDecoder();

    @Override
    public Command decodeNextByte(byte nextByte) {
        String username = stringEncoderDecoder.decodeNextByte(nextByte);
        if (username != null) {
            return new LoginRequest(username);
        }
        return null;
    }
}

File: java to c++ builders/LoginRequestCommandBuilder.cpp
#include "LoginRequestCommandBuilder.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace CommandBuilders
					{
						using bgu::spl171::net::impl::TFTP::Commands::Command;
						using bgu::spl171::net::impl::TFTP::Commands::LoginRequest;
						using bgu::spl171::net::impl::TFTP::Commands::ReadRequest;
						using bgu::spl171::net::impl::TFTP::EncoderDecoders::StringEncoderDecoder;

						Command *LoginRequestCommandBuilder::decodeNextByte(char nextByte)
						{
							std::wstring username = stringEncoderDecoder->decodeNextByte(nextByte);
							if (username != L"")
							{
								return new LoginRequest(username);
							}
							return nullptr;
						}
					}
				}
			}
		}
	}
}
